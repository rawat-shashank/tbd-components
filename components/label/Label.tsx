import React from 'react';
import { StyledLabel } from './Label.styles';
import { LabelProps } from './Label.types';

/**
 * Display a label.
 *
 * >Label component comes with default styles which can be overriden
 * by using either classname or style.
 */
export const Label = ({ label, className, style, disabled }: LabelProps) => (
  <StyledLabel
    className={className}
    style={style}
    disabled={disabled}
    data-testid="label"
  >
    {label}
  </StyledLabel>
);
