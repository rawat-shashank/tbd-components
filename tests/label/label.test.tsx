import { render } from '@testing-library/react';
import React from 'react';
import { Default, Disabled } from '../../stories/label/Label.stories';
import { ThemeProvider } from 'styled-components';

describe('Unit Test | Label Component', () => {
  it('should match label', () => {
    const renderComponent = () => render(<Default {...Default.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveStyle({ color: '#414141' });
    expect(testComponent).toHaveTextContent(Default.args.label);
  });

  it('should have attribute disable', () => {
    const renderComponent = () => render(<Disabled {...Disabled.args} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveStyle({ color: '#A0A0A0' });
    expect(testComponent).toBeDisabled();
  });

  it('should override with theme color', () => {
    const theme: any = { text: { secondary: 'red' } };
    const renderComponent = () =>
      render(
        <ThemeProvider theme={theme}>
          <Default {...Default.args} />
        </ThemeProvider>
      );
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveStyle({ color: theme.text.secondary });
  });

  it('should override with theme disabled color', () => {
    const theme: any = { icon: { disabled: 'black' } };
    const renderComponent = () =>
      render(
        <ThemeProvider theme={theme}>
          <Disabled {...Disabled.args} />
        </ThemeProvider>
      );
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveStyle({ color: theme.icon.disabled });
  });

  it('should work with external className', () => {
    const renderComponent = () =>
      render(<Default {...Default.args} className="test" />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveClass('test');
  });

  it('should work with external style', () => {
    const style = {
      color: 'red',
    };
    const renderComponent = () =>
      render(<Default {...Default.args} style={style} />);
    const { getByTestId } = renderComponent();
    const testComponent = getByTestId('label');
    expect(testComponent).toHaveStyle({ color: style.color });
  });

  it('should match snapshot', () => {
    const { container } = render(<Default {...Default.args} />);
    expect(container).toMatchSnapshot();
  });
});
