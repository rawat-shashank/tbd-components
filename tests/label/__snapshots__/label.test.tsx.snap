// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test | Label Component should match snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL fYlwKI"
    data-testid="label"
  >
    Label
  </button>
</div>
`;
